sentry-apache-http-client-5/src/main/java/io/sentry/transport/apache/ApacheHttpClientTransport.java

```diff
@@ -0,0 +1,147 @@
+import org.apache.hc.client5.http.impl.async.CloseableHttpAsyncClient;
+import org.apache.hc.core5.concurrent.FutureCallback;
+import org.apache.hc.core5.http.ContentType;
+import org.apache.hc.core5.http.Header;
+import org.apache.hc.core5.io.CloseMode;
+import org.apache.hc.core5.util.TimeValue;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * {@link ITransport} implementation that executes request asynchronously in a non-blocking manner
+ * using Apache Http Client 5.
+ */
+public final class ApacheHttpClientTransport implements ITransport {
+  private final @NotNull SentryOptions options;
+  private final @NotNull RequestDetails requestDetails;
+  private final @NotNull CloseableHttpAsyncClient httpclient;
+  private final @NotNull RateLimiter rateLimiter;
+  private final @NotNull AtomicInteger currentlyRunning;
+
+  public ApacheHttpClientTransport(
+      final @NotNull SentryOptions options,
+      final @NotNull RequestDetails requestDetails,
+      final @NotNull CloseableHttpAsyncClient httpclient,
+      final @NotNull RateLimiter rateLimiter) {
+    this(options, requestDetails, httpclient, rateLimiter, new AtomicInteger());
+  }
+
+  ApacheHttpClientTransport(
+      final @NotNull SentryOptions options,
+      final @NotNull RequestDetails requestDetails,
+      final @NotNull CloseableHttpAsyncClient httpclient,
+      final @NotNull RateLimiter rateLimiter,
+      final @NotNull AtomicInteger currentlyRunning) {
+    this.options = Objects.requireNonNull(options, "options is required");
+    this.requestDetails = Objects.requireNonNull(requestDetails, "requestDetails is required");
+    this.httpclient = Objects.requireNonNull(httpclient, "httpclient is required");
+    this.rateLimiter = Objects.requireNonNull(rateLimiter, "rateLimiter is required");
+    this.currentlyRunning =
+        Objects.requireNonNull(currentlyRunning, "currentlyRunning is required");
+    this.httpclient.start();
+  }
+
+  @Override
+  @SuppressWarnings("FutureReturnValueIgnored")
+  public void send(SentryEnvelope envelope, Object hint) throws IOException {
+    if (isSchedulingAllowed()) {
+      final SentryEnvelope filteredEnvelope = rateLimiter.filter(envelope, hint);
+
+      if (filteredEnvelope != null) {
+        currentlyRunning.incrementAndGet();
+
+        try (final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+            final GZIPOutputStream gzip = new GZIPOutputStream(outputStream)) {
+          options.getSerializer().serialize(filteredEnvelope, gzip);
+
+          final SimpleHttpRequest request =
+              SimpleHttpRequests.post(requestDetails.getUrl().toString());
+          request.setBody(outputStream.toByteArray(), ContentType.APPLICATION_JSON);
+          request.setHeader("Content-Encoding", "gzip");
+          request.setHeader("Content-Type", "application/x-sentry-envelope");
+          request.setHeader("Accept", "application/json");
```

Comment:
Should this be part of RequestDetailsResolver or even extracting to a new class? it's a duplicated code right now.

Reply:
If it's just 3 lines of code I wouldn't bother

Reply:
if `RequestDetailsResolver` is used in both transports and they are meant to configure the connection with its headers, I don't see why duplicating it though.
https://github.com/getsentry/sentry-java/blob/gh-1097-apache-non-blocking-transport/sentry/src/main/java/io/sentry/transport/HttpConnection.java#L124-L126

Reply:
It's up to the transport to decide if it sends gzip or not, I also don't think that we should set fixed accept header. `RequestDetailsResolver` returns headers that users MUST send to Sentry.